lib/puppet/transaction/report.rb

```diff
@@ -158,6 +168,21 @@ def compute_status(resource_metrics, change_metric)
   end
 
   # @api private
+  def has_noop_change_events?(resource)
+    resource.events.any? { |event| (event.status == 'noop' && event.name == :content_changed) }
```

Comment:
Point in case, this is a mode change on file (in yaml):

```
    events:
    - !ruby/object:Puppet::Transaction::Event
      audited: false
      property: mode
      previous_value: '0644'
      desired_value: '777'
      historical_value: '0644'
      message: mode changed '0644' to '0777'
      name: :mode_changed
      status: success
      time: 2016-06-07 11:05:46.888644000 +01:00
```

Notice it is `:mode_changed`, not `:content_changed`.